### PR TITLE
Fix marshaling of ByVal strings

### LINKED ISSUE #1741: null terminated string (Pinvoke)
```
<DllImport("user32.dll", EntryPoint:="EnumDisplayDevices", CharSet:=CharSet.Auto, CallingConvention:=CallingConvention.StdCall)>
        Public Shared Function EnumDisplayDevices(lpDevice As String, iDevNum As UInteger, ByRef lpDisplayDevice As DISPLAY_DEVICE, dwFlags As UInteger) As Integer
        End Function
Dim c_Dev As New DISPLAY_DEVICE With {.cb = Marshal.SizeOf(c_Dev)}
EnumDisplayDevices(Nothing, 0, c_Dev, 0)
Dim p_Adapter As String = c_Dev.DeviceString

```
NET 5 return correct string (CharSet:=CharSet.Auto)
AOT return string + Chr(0),Chr(0),Chr(0),Chr(0),Chr(0),.........(CharSet:=CharSet.Auto)
Any other CharSet return nothing (AOT)

All day long I couldn't figure out what was wrong...:))
I noticed it only in the log file with enabled all system chars

Marshal issue?
Perhaps the problem is deeper and not just here
Yep, .TrimEnd(Chr(0), Chr(32)) fix issue...

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #1741

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs [hunk 1]
```diff
@@ -55,54 +55,55 @@ public static unsafe string UTF8StringToString(byte* buffer)
...
+            else
+            {
+                (*pNative) = (byte)'\0';
+            }
         }
 
         public static unsafe string ByValAnsiStringToString(byte* buffer, int length)
         {
-            return PInvokeMarshal.ByValAnsiStringToString(buffer, length);
+            int end = SpanHelpers.IndexOf(ref *(byte*)buffer, 0, length);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs:78]: Copied CoreCLR logic for all these.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/PInvokeMarshal.cs [hunk 2]
```diff
@@ -418,58 +416,6 @@ public static unsafe string AnsiStringToString(byte* pchBuffer)
...
-        /// <remarks>
-        /// * This version works with array instead string, it means that there will be NO NULL to terminate the array.
-        /// * The buffer to store the byte array must be allocated by the caller and must fit managedArray.Length.
-        /// </remarks>
-        /// <param name="managedArray">UNICODE wide char array</param>
-        /// <param name="pNative">Allocated buffer where the ansi characters must be placed. Could NOT be null. Buffer size must fit char[].Length.</param>
-        /// <param name="expectedCharCount">Minimum expected length of managedArray</param>
-        /// <param name="bestFit">Enable best-fit mapping behavior</param>
-        /// <param name="throwOnUnmappableChar">Throw an exception on an unmappable Unicode character</param>
-        public static unsafe void ByValWideCharArrayToAnsiCharArray(char[] managedArray, byte* pNative, int expectedCharCount,
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/PInvokeMarshal.cs:433]: Unused MCG helpers

File: src/tests/nativeaot/SmokeTests/PInvoke/PInvokeNative.cpp [hunk 3]
```diff
@@ -572,7 +572,8 @@ DLL_EXPORT bool __stdcall InlineArrayTest(inlineStruct* p, inlineUnicodeStruct *
     if (CompareAnsiString(p->inlineString, "Hello") != 1)
        return false;
 
-    if (!VerifyUnicodeString(q->inlineString))
+    unsigned short expected[] = { 'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 0 };
```
Comment (jkotas) [inline src/tests/nativeaot/SmokeTests/PInvoke/PInvokeNative.cpp:575]: The test was validating wrong behavior. The test failed on CoreCLR before this change.

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**